### ISSUE TITLE
macOS: Missing UI delegate for WebView (no alerts, <input type="file"> dialog)

### ISSUE THREAD

**Issue by denisw** [neutralinojs/neutralinojs#1014]
**Describe the bug**

WebKit's [WKWebView](https://developer.apple.com/documentation/webkit/wkwebview?language=objc) needs an implementation of [WKUIDelegate](https://developer.apple.com/documentation/webkit/wkuidelegate?language=objc) for some interactions that require native UI bits, such as:

- JS alerts and prompts (`alert()`, `confirm()`, `prompt()`)
- File dialogs, e.g. when clicking on an `<input type="file">`

Neutralino does not provide such a delegate (see `webview.h`). Accordingly, the above browser features do not work on macOS (nothing happens).

**To reproduce**

1. Create a new Neutralino project.
2. Add `<input type="file">` to the HTML.
3. Add `alert('foo')` call to the JavaScript code.
4. Run the app with `neu run`.

**Expected behavior**

- I can see a dialog saying "foo" when I start the app.
- Clicking on the file input opens a file dialog.

**Actual behavior**

- No dialog is shown.
- Clicking on the file input does nothing (but dragging a file from Finder to the file input works).

**Specifications** 

 - OS: macOS 12.6 arm64
 - Neutralinojs version: v4.8.0
 - Neutralinojs client library version: v3.7.0
 - Neutralinojs CLI version: v9.3.1

**Additional context**

Follow-up of #1005.

**Comment by JannikGM** [comment]
> File dialogs, e.g. when clicking on an `<input type="file">`

I only tested on macOS and assumed this was by design.
I personally abstracted this away in our app like this:

```js
      Neutralino.os.showOpenDialog('Open a file', {
        multiSelections: false
      }).then((entries) => {
        const path = entries[0];
        Neutralino.filesystem.readBinaryFile(path).then((data) => {
          const dataArray = new Uint8Array(data);
          callback(dataArray, path);
        });
      })
      return;
```

I added similar code for downloading / saving a file.

**Comment by denisw** [comment]
I don’t think it is by design because these features work out of the box on both Windows and Linux. I assume it is just an oversight in the macOS implementation, which requires manual wiring code for these things to work (the mentioned UI delegate) while the other platforms’ webview libraries don’t.
